runtime/gear/src/tests.rs

```diff
@@ -17,116 +17,217 @@
+    check_instruction_weight(weights.i64gts, 4_000);
+    check_instruction_weight(weights.i32gts, 2_000);
+    check_instruction_weight(weights.i64gtu, 4_000);
+    check_instruction_weight(weights.i32gtu, 2_000);
+    check_instruction_weight(weights.i64les, 4_000);
+    check_instruction_weight(weights.i32les, 2_000);
+    check_instruction_weight(weights.i64leu, 4_000);
+    check_instruction_weight(weights.i32leu, 2_000);
+
+    check_instruction_weight(weights.i64ges, 4_000);
+    check_instruction_weight(weights.i32ges, 2_000);
+    check_instruction_weight(weights.i64geu, 4_000);
+    check_instruction_weight(weights.i32geu, 2_000);
+    check_instruction_weight(weights.i64add, 2_500);
+    check_instruction_weight(weights.i32add, 1_000);
+    check_instruction_weight(weights.i64sub, 3_000);
+    check_instruction_weight(weights.i32sub, 1_000);
+    check_instruction_weight(weights.i64mul, 4_000);
+    check_instruction_weight(weights.i32mul, 2_300);
+    check_instruction_weight(weights.i64divs, 4_800);
+    check_instruction_weight(weights.i32divs, 3_800);
+
+    check_instruction_weight(weights.i64divu, 5_200);
+    check_instruction_weight(weights.i32divu, 4_200);
+    check_instruction_weight(weights.i64rems, 21_100);
+    check_instruction_weight(weights.i32rems, 15_100);
+    check_instruction_weight(weights.i64remu, 4_300);
+    check_instruction_weight(weights.i32remu, 4_300);
+    check_instruction_weight(weights.i64and, 3_000);
+    check_instruction_weight(weights.i32and, 1_000);
+    check_instruction_weight(weights.i64or, 3_000);
+    check_instruction_weight(weights.i32or, 1_000);
+    check_instruction_weight(weights.i64xor, 3_000);
+    check_instruction_weight(weights.i32xor, 1_000);
+
+    check_instruction_weight(weights.i64shl, 2_500);
+    check_instruction_weight(weights.i32shl, 1_000);
+    check_instruction_weight(weights.i64shrs, 2_500);
+    check_instruction_weight(weights.i32shrs, 1_000);
+    check_instruction_weight(weights.i64shru, 2_500);
+    check_instruction_weight(weights.i32shru, 1_000);
+    check_instruction_weight(weights.i64rotl, 2_000);
+    check_instruction_weight(weights.i32rotl, 1_000);
+    check_instruction_weight(weights.i64rotr, 2_500);
+    check_instruction_weight(weights.i32rotr, 1_000);
+}
+
+#[test]
+fn page_costs_heuristic_test() {
+    let page_costs: PageCosts = MemoryWeights::<Runtime>::default().into();
+    let expected = PageCosts {
+        lazy_pages_signal_read: 28_000_000.into(),
+        lazy_pages_signal_write: 33_000_000.into(),
+        lazy_pages_signal_write_after_read: 8_600_000.into(),
+        lazy_pages_host_func_read: 29_000_000.into(),
+        lazy_pages_host_func_write: 33_000_000.into(),
+        lazy_pages_host_func_write_after_read: 7_500_000.into(),
+        load_page_data: 8_700_000.into(),
+        upload_page_data: 104_000_000.into(),
+        static_page: 100.into(),
+        mem_grow: 100.into(),
+        parachain_load_heuristic: 0.into(),
+    };
```

Comment:
Can't we go without manually constructing struct? As it did in test above

Reply:
No better make like in my test in test above. Because in case somebody will add new fields in this structs then this test s won't compile